tests/treemap/test_py_pkg_treemap.py

```
@@ -949,341 +882,199 @@ def test_collect_coverage_data_subprocess(
-        with (  # For subprocess scenarios, mock everything
+        with (
             patch("subprocess.run") as mock_run,
             patch("tempfile.NamedTemporaryFile") as mock_tempfile,
             patch("builtins.open", create=True),
             patch("os.path.exists", return_value=True),
             patch("os.unlink"),
-            patch("json.load") as mock_json_load,
+            patch("json.load", return_value=coverage_data),
         ):
+            mock_tempfile.return_value.__enter__.return_value.name = (
+                "/tmp/coverage.json"
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/coverage.json"

(S108)